src/System.IO.FileSystem/tests/FSAssert.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    public static class FSAssert
+    {
+        public static void ThrowsSharingViolation(Action testCode)
+        {
+            Exception e = Assert.ThrowsAny<Exception>(testCode);
+            Assert.True(e is UnauthorizedAccessException || e is IOException, "Exception should be either UnauthorizedAccessException or IOException: " + e.ToString());
+        }
+
+        public static void CompletesSynchronously(Task task, bool checkCompletedSynchronously = true)
+        {
+            IAsyncResult result = (IAsyncResult)task;
+
+            // some IAsyncResult implementations don't set CompletedSynchronously
+            // if that is expected, relax the check to just expect completion
+            // this does allow for a race condition (task completes asynchronously
+            // before we are able to check) but that is all that we can do.
+            bool completed = checkCompletedSynchronously ? !result.CompletedSynchronously : !result.IsCompleted;
```

Comment:
That's unfortunate.  Most of the time I am using this is just to make sure we get an exception thrown synchronously so I never get here.  Maybe I should just remove the IAsyncResult check and just leave the comment about the race.